clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp

```diff
@@ -42,6 +42,14 @@ static RValue buildLibraryCall(CIRGenFunction &CGF, const FunctionDecl *FD,
   return CGF.buildCall(E->getCallee()->getType(), callee, E, ReturnValueSlot());
 }
 
+template <class Operation>
+static RValue buildUnaryFPBuiltin(CIRGenFunction &CGF, const CallExpr &E) {
+  auto Arg = CGF.buildScalarExpr(E.getArg(0));
+  auto Call =
+      CGF.getBuilder().create<Operation>(Arg.getLoc(), Arg.getType(), Arg);
+  return RValue::get(Call->getResult(0));
+}
+
```

Comment:
To keep codegen parity with the original method and ensure we crash when running constrained FP operations:

```suggestion
template <class Operation>
static RValue buildUnaryMaybeConstrainedFPBuiltin(CIRGenFunction &CGF, const CallExpr &E) {
  auto Arg = CGF.buildScalarExpr(E.getArg(0));

  CIRGenFunction::CIRGenFPOptionsRAII FPOptsRAII(CGF, &E);
  if (CGF.getBuilder().getIsFPConstrained()) {
    llvm_unreachable("constraint FP operations are NYI");
  } else {
    auto Call =
        CGF.getBuilder().create<Operation>(Arg.getLoc(), Arg.getType(), Arg);
    return RValue::get(Call->getResult(0));
  }
}
```